libcontainer/configs/validate/validator.go

```diff
@@ -82,6 +83,13 @@ func hostname(config *configs.Config) error {
 	return nil
 }
 
+func domainname(config *configs.Config) error {
+	if config.Domainname != "" && !config.Namespaces.Contains(configs.NEWUTS) {
```

Comment:
Perhaps it's better to merge this check together with the hostname one (and have the unified error).

Reply:
I'd rather combine the checks. Something like

```go
if config.Hostname = "" && config.Domainname = "" {
	return nil
}
if !config.Namespaces.Contains(configs.NEWUTS) {
	return errors.New("unable to set host/domain name without a private UTS namespace")
}
```